lib/net/peer.js

```diff
@@ -242,10 +258,12 @@ class Peer extends EventEmitter {
 
     this.socket = socket;
 
-    this.brontide.once('error', (err) => {
-      this.error(err);
-      this.destroy();
-    });
+    if (this.encrypt) {
+      this.brontide.once('error', (err) => {
+        this.error(err);
+        this.destroy();
+      });
+    }
```

Comment:
I'm not sure you need to `if` this - looking at the constructor, every peer still has a `this.brontide = BrontideStream()` whether or not it gets used.

Reply:
I don't think there is a need to set up listeners when they will never be used

Reply:
I guess it just seems cleaner, but I don't care enough to push it. I just think the fact that the instance still has a `BrontideStream` means that it is in use, and if it's ever opened by some external caller, it's not going to _hurt_ to have an error listener.

Reply:
Event listeners do add some resource usage and can result in memory leaks if they are not removed

Reply:
> I guess it just seems cleaner

I do agree that it looks way cleaner and adds less mental overhead when reasoning about the codepath.

Reply:
If trying to connect to the brontide server when using the plaintext protocol, this should definitely keep the listener. Need to verify that.

Reply:
Verified that we should set the listener in case somebody tries to connect in clear on the brontide port. See https://github.com/handshake-org/hsd/pull/331#issuecomment-577457042